proxy.js

```diff
@@ -171,4 +171,4 @@
 
   scope.Proxy['revocable'] = scope.Proxy.revocable;
   scope['Proxy'] = scope.Proxy;
-})(typeof module !== 'undefined' && module['exports'] ? global : window);
+})(typeof process !== "undefined" && {}.toString.call(process) === "[object process]" ? global : typeof WorkerGlobalScope !== 'undefined' && self instanceof WorkerGlobalScope ? self : window);
```

Comment:
Detecting module.exports will give you false positives for non-Node based commonjs runtimes (like SystemJS).  The process method is the best I've seen because you'd have to override Object.prototype.toString to make this truthy; in which case you are probably trying to make it seem as the code is running in Node.

You could also just detect `global` but that's often faked as well, but that would be slightly better.  Soonish we'll have [System.global](https://github.com/tc39/proposal-global) and this won't be as much of a problem.